### ISSUE TITLE
Super issue

### ISSUE THREAD

**Issue by Apollo3zehn** [Apollo3zehn/HDF5.NET#9]
# before alpha release
- [ ] Parallel testing support or run tests sequentially
- [ ] Repair Little/Big Endian conversion
- [ ] Check if generic reading requires changes to hyperslab selections
- [ ] Ensure that buffer size is checked before copy
- [ ] Improve File.Open signature

# before beta release
- [ ] Parallel read support
- [ ] Read unknown structs using ExpandoObject
- [ ] Linq To Object Support 
- [ ] Read multidim array: with implicity conversion operator? Or with generic Array[] constraint and TReturn (not possible yet)?

## missing tests
Tests
- [ ] skip filter
- [ ] do not filter edge chunks
- [ ] add tests with max dims != dims

**Comment by Ams627** [comment]
Does this mean that currently, it is impossible to read variable length non string data types?

**Comment by Apollo3zehn** [comment]
Yes, that is not possible yet. Mainly because it not know how variable length data would look like. Do you have an example?
